src/platform_impl/linux/x11/event_processor.rs

```diff
@@ -869,12 +869,27 @@ impl EventProcessor {
         // Set the timestamp.
         self.target.xconn.set_timestamp(xev.time as xproto::Timestamp);
 
+        // A key input implies having the focus
         let window = match self.active_window {
             Some(window) => window,
-            None => return,
+            None => *self.active_window.insert(xev.window as xproto::Window),
         };
 
         let window_id = mkwid(window);
+        let focused = self.with_window(window, |window| {
+            let mut shared_state_lock = window.shared_state_lock();
+            if !shared_state_lock.has_focus {
+                shared_state_lock.has_focus = true;
+                true
+            } else {
+                false
+            }
+        });
+        if focused == Some(true) {
+            let event = Event::WindowEvent { window_id, event: WindowEvent::Focused(true) };
+            callback(&self.target, event);
+        }
```

Comment:
Should only send focus event when we actually will be sending anything related to the keyboard input.

Reply:
Even if this particular keyboard event does not end in notifying the user of anything, my argument was that if I'm getting this X11 event here it is because this window has the focus. And if `shared_state_lock.has_focus` is false, then I should already have received it, so I'm sending it now to compensate for the missing one.